res/qml/LibraryControl.qml

```diff
@@ -0,0 +1,188 @@
+
+    signal moveSelection(int offset)
+    signal loadSelectedTrack(string group, bool play)
+    signal loadSelectedTrackIntoNextAvailableDeck(bool play)
+
+    FocusedWidgetControl {
+        id: focusedWidgetControl
+
+        Component.onCompleted: this.value = FocusedWidgetControl.WidgetKind.LibraryView
+    }
+
+    Mixxx.ControlProxy {
+        group: "[Library]"
+        key: "GoToItem"
+        onValueChanged: {
+            if (value != 0 && root.focusWidget == FocusedWidgetControl.WidgetKind.LibraryView)
+                root.loadSelectedTrackIntoNextAvailableDeck(false);
+
+        }
+    }
+
+    Mixxx.ControlProxy {
+        group: "[Playlist]"
+        key: "LoadSelectedIntoFirstStopped"
+        onValueChanged: {
+            if (value != 0 && root.focusWidget == FocusedWidgetControl.WidgetKind.LibraryView)
+                root.loadSelectedTrackIntoNextAvailableDeck(false);
+
+        }
+    }
+
+    Mixxx.ControlProxy {
+        group: "[Playlist]"
+        key: "SelectTrackKnob"
+        onValueChanged: {
+            if (value != 0) {
+                root.focusWidget = 3;
+                root.moveSelection(value);
+            }
+        }
+    }
+
+    Mixxx.ControlProxy {
+        group: "[Playlist]"
+        key: "SelectPrevTrack"
+        onValueChanged: {
+            if (value != 0) {
+                root.focusWidget = 3;
+                root.moveSelection(-1);
+            }
+        }
+    }
+
+    Mixxx.ControlProxy {
+        group: "[Playlist]"
+        key: "SelectNextTrack"
+        onValueChanged: {
+            if (value != 0) {
+                root.focusWidget = 3;
+                root.moveSelection(1);
+            }
+        }
+    }
+
+    Mixxx.ControlProxy {
+        group: "[Library]"
+        key: "MoveVertical"
+        onValueChanged: {
+            if (value != 0 && root.focusWidget == FocusedWidgetControl.WidgetKind.LibraryView)
+                root.moveSelection(value);
+
+        }
+    }
+
+    Mixxx.ControlProxy {
+        group: "[Library]"
+        key: "MoveUp"
+        onValueChanged: {
+            if (value != 0 && root.focusWidget == FocusedWidgetControl.WidgetKind.LibraryView)
+                root.moveSelection(-1);
+
+        }
+    }
+
+    Mixxx.ControlProxy {
+        group: "[Library]"
+        key: "MoveDown"
+        onValueChanged: {
+            if (value != 0 && root.focusWidget == FocusedWidgetControl.WidgetKind.LibraryView)
+                root.moveSelection(1);
+
+        }
+    }
+
+    Mixxx.ControlProxy {
+        id: numDecksControl
+
+        group: "[Master]"
+        key: "num_decks"
+    }
+
+    Instantiator {
+        model: numDecksControl.value
+
+        delegate: LibraryLoadSelectedTrackControl {
+            group: "[Channel" + (index + 1) + "]"
+            enabled: root.focusWidget == FocusedWidgetControl.WidgetKind.LibraryView
+            onLoadTrackRequested: root.loadSelectedTrack(group, play)
+        }
+
+    }
```

Comment:
[`Note: Instantiator is part of QtQml.Models since version 2.14 and part of QtQml since version 2.1. Importing Instantiator via QtQml is deprecated since Qt 5.14.`](https://doc.qt.io/qt-5/qml-qtqml-models-instantiator.html#details)
What was the reason for using an `Instantiator` vs a `Repeater`?
(I don't understand the difference really, I think `Repeater` is used more often in Qt (at least in their examples).)

Reply:
Repeater only works with `Item` (which has width/height). Instantiator also works with basic Components.

Reply:
Right, but we're using an `Item` here, so why did you choose the `Instantiator` over the `Repeater`?

Reply:
Weirdly enough, mixxx crashes when I use `QtObject` instead of `Item`. I guess some bug in Qt 5.

Reply:
Mhmm ok. then I'd prefer if that was documented somewhere so we can reason about this design decision later.